File: Python/724. Find Pivot Index.py
from typing import List

# noinspection PyMethodMayBeStatic
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        left, right = 0, sum(nums)

        for idx, val in enumerate(nums):
            right -= val

            if left == right:
                return idx

            left += val
        return -1

File: PHP/724. Find Pivot Index.php
<?php

/**
 * @noinspection PhpIllegalPsrClassPathInspection
 * @noinspection PhpMultipleClassDeclarationsInspection
 * @noinspection PhpUnused
 */
class Solution
{
    /**
     * @param  int[]  $nums
     */
    public function pivotIndex(array $nums): int
    {
        $left = 0;
        $right = array_sum($nums);

        for ($i = 0; $i < count($nums); $i++) {
            $current = $nums[$i];
            if ($left === $right -= $current) {
                return $i;
            }
            $left += $current;
        }

        return -1;
    }
}
